### PR TITLE
Added subfund tests, moved monetize to utils file

### PR DESCRIPTION
**Pull request by JakeDawkins**

### CODE REVIEW

**Comment by jbaxleyiii** [comment]
**[Storybook Hub](https://hub.getstorybook.io/)** is building storybooks for every commit in this PR.

**Latest Storybook: https://ae6e1e7f-c119-4ab3-905a-f3b5abac845c.sbook.io/**
<sub>( Visit [here](https://hub.getstorybook.io/app/github.com/NewSpring/Holtzmann/branch/test-cart-subfunds/) for more info )</sub>

> _This comment is automatically added by Storybook Hub_

File: imports/util/format/currency.js [hunk 1]
```diff
@@ -1,11 +1,41 @@
...
 export default toCurrency;
+
+/*
+
+  monetize
+  @param value: the dollar value to format and place a $ in front of
+  @param fixed: whether or not to enforce decimal places
+
+*/
+const monetize = (value: string | number, fixed: boolean): string => {
```
Comment (delianides) [inline imports/util/format/currency.js]: `fixed` needs to be optional. 

```js
const monetize = (value: string | number, fixed?: boolean): string => {
```

File: imports/blocks/add-to-cart/Layout.js [hunk 2]
```diff
@@ -5,7 +5,7 @@ import Forms from "../../components/forms";
 
 import CheckoutButtons from "../checkout-buttons";
 
-import SubFund from "./Subfund";
+import SubFund from "./Subfund"; //eslint-disable-line
```
Comment (jbaxleyiii) [inline imports/blocks/add-to-cart/Layout.js]: why is this disabled?

File: imports/blocks/add-to-cart/Subfund/Layout.js [hunk 3]
```diff
@@ -1,10 +1,74 @@
...
+};
+
+const RenderAmountInput = ({
+  state,
+  classes,
+  format,
+  preFill,
+  inputVal,
+}: IRenderAmountInput) => {
+  if (state.fund) {
```
Comment (jbaxleyiii) [inline imports/blocks/add-to-cart/Subfund/Layout.js]: This is semantic but I'd prefer early returns when possible.

i.e 
```
if (!state.fund) return null;
return (/* the rest */);
```

File: imports/blocks/add-to-cart/Subfund/__tests__/Layout.js [hunk 4]
```diff
@@ -0,0 +1,121 @@
+
+import { mount } from "enzyme";
+import { mountToJson } from "enzyme-to-json";
+import { reset, startBuffering } from "aphrodite/lib/inject";
+import { getSingleSpecWrapper } from "../../../../util/tests/data-spec.js";
+
+import Layout from "../Layout";
+
+const generateComponent = (additionalProps={}) => {
```
Comment (jbaxleyiii) [inline imports/blocks/add-to-cart/Subfund/__tests__/Layout.js]: change spacing to:
`const generateComponent = (additionalProps = {}) => {`

The spacing around the `=`

File: imports/util/format/currency.js [hunk 5]
```diff
@@ -1,11 +1,41 @@
...
+
+  monetize
+  @param value: the dollar value to format and place a $ in front of
+  @param fixed: whether or not to enforce decimal places
+
+*/
+const monetize = (value: string | number, fixed?: boolean): string => {
+  let amount = typeof value === "number" ? `${value}` : value;
+
+  if (!amount || !amount.length) {
```
Comment (jbaxleyiii) [inline imports/util/format/currency.js]: lets one line this:

`if (!amount || !amount.length) return "$0.00";`

**Comment by richarddubay** [comment]
@jbaxleyiii I can make these changes to get this merged. There is still a good bit of refactoring/typing that needs to be done, but I think it would make me happier to do it on a different PR. Is that okay? Also, there is a different PR (https://github.com/NewSpring/Holtzman/pull/1434) open to be merged into this branch. It's marked as [WIP], but might be worth taking a look at before merging this one?

File: imports/blocks/add-to-cart/Layout.js (hunk 2 above)
Comment (jbaxleyiii) [inline imports/blocks/add-to-cart/Layout.js]: ah, 👍 lets leave as is

**Comment by jbaxleyiii** [comment]
@richarddubay I think that's a great idea (the other branches). I think we can merge this in and update the base branch on the other PR

**Comment by richarddubay** [comment]
@jbaxleyiii this is ready to merge whenever you are ready to merge it.

**Pull request merged**